tests/unit-core/test-promise.c

```diff
@@ -158,6 +174,34 @@ main (void)
   jerry_resolve_or_reject_promise (my_promise2, str_resolve, true);
   jerry_resolve_or_reject_promise (my_promise1, str_reject, false);
 
+  jerry_value_t number = jerry_create_number (42);
+
+  jerry_char_t src_3[] = "Promise.reject(42)";
+  jerry_value_t my_promise3 = jerry_eval (src_3, sizeof (src_3) - 1, JERRY_PARSE_NO_OPTS);
+  TEST_ASSERT (!jerry_value_is_error (my_promise3));
+  TEST_ASSERT (tracter_counter == 1);
+  TEST_ASSERT (jerry_get_boolean_value (jerry_binary_operation (JERRY_BIN_OP_STRICT_EQUAL,
+                                                                last_rejected_promise,
+                                                                my_promise3)));
+  TEST_ASSERT (jerry_get_boolean_value (jerry_binary_operation (JERRY_BIN_OP_STRICT_EQUAL,
+                                                                last_rejected_reason,
+                                                                number)));
+  jerry_release_value (last_rejected_promise);
+  jerry_release_value (last_rejected_reason);
+  jerry_release_value (number);
+
+  jerry_char_t src_4[] = "Promise.race()";
+  jerry_value_t my_promise4 = jerry_eval (src_4, sizeof (src_4) - 1, JERRY_PARSE_NO_OPTS);
+  TEST_ASSERT (!jerry_value_is_error (my_promise4));
+  TEST_ASSERT (tracter_counter == 2);
+  TEST_ASSERT (jerry_get_boolean_value (jerry_binary_operation (JERRY_BIN_OP_STRICT_EQUAL,
+                                                                last_rejected_promise,
+                                                                my_promise4)));
+  TEST_ASSERT (jerry_value_is_object (last_rejected_reason));
+  TEST_ASSERT (jerry_get_error_type (last_rejected_reason) == JERRY_ERROR_TYPE);
+  jerry_release_value (last_rejected_promise);
+  jerry_release_value (last_rejected_reason);
+
```

Comment:
There should be also a test where the "global" reject handler is not invoked as the reject was handled already.